Sources/Java2SwiftLib/ConfigFileGenerator.swift

```diff
@@ -0,0 +1,257 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import JavaKit
+import JavaKitJar
+
+/// Java2Swift Config file generation tool
+package struct ConfigFileGenerator {
+  /// The module name to used for generation
+  let moduleName: String
+  /// the specified output directory
+  let outputDirectory: String?
+  /// if it's handling jar file
+  let jar: Bool
+  /// Whether we have ensured that the output directory exists.
+  private var createdOutputDirectory: Bool = false
+  private var inNewSwiftPackage: Bool
+
+  package init(moduleName: String,
+              outputDirectory: String? = nil,
+              jar: Bool,
```

Comment:
I'd prefer we avoid just bools like this that are unclear what behavior changes they cause like this. Instead the caller should construct a correct output directory, which is the only thing this was used for.

Reply:
Yeah. I followed this from `JavaToSwift`.  Any suggestions on this? We can update these together.